src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -173,10 +178,42 @@ private async Task<bool> TryPopulateNewJobFields()
         {
             // Build query based on new search params
             // Find supported, but not yet searchable params
-            var notYetIndexedParams = _supportedSearchParameterDefinitionManager.GetSearchParametersRequiringReindexing();
+            var possibleNotYetIndexedParams = _supportedSearchParameterDefinitionManager.GetSearchParametersRequiringReindexing();
+            _notYetIndexedParams = new List<SearchParameterInfo>();
+
+            var resourceList = new HashSet<string>();
+
+            // filter list of SearchParameters by the target resource types
+            if (_reindexJobRecord.TargetResourceTypes.Any())
+            {
+                foreach (var searchParam in possibleNotYetIndexedParams)
+                {
+                    var searchParamResourceTypes = GetDerivedResourceTypes(searchParam.BaseResourceTypes);
+                    var matchingResourceTypes = searchParamResourceTypes.Intersect(_reindexJobRecord.TargetResourceTypes);
+                    if (matchingResourceTypes.Any())
```

Comment:
I am not sure if that is a warning, as it is the goal of specifying target types in the reindex job.  The user wants to avoid reindexing certain searchparameters, at least in some scenarios.  But more info is good, I will add an info log.